.github/workflows/clients.yml

```diff
@@ -0,0 +1,124 @@
+name: Clients
+
+concurrency:
+  group: clients-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+
+jobs:
+  client-dotnet:
+    strategy:
+      matrix:
```

Comment:
We might want to sprinkle `max-parallel:` around the matrix jobs here - so that a single commit / run doesn't use all the runners available to us.